### ISSUE TITLE
Mail gets marked as unread when moving between labels

### ISSUE THREAD

**Issue by bansalsohil** [nylas/N1#591]
When you move mail from one label to another. It gets marked as unread.

**Comment by bengotow** [comment]
Hi — what mail provider / operating system are you using?

**Comment by enoversum** [comment]
Ah well, this is the dreadful thing I already reported. Once I sent and archived - or only one of the actions - an e-mail, it turns up unread again, often pops back into my inbox. Already reported it in, using a classic IMAP connection (is everybody using Gmail these days?).

**Comment by enoversum** [comment]
Just to report in: If I take a long-read e-mail, open it, move it to a different folder, it disappears, while N1 tells me it is syncing. A few seconds later, it pops up in the inbox again as unread. I can reliably reproduce it. IMAP in use.
Even stranger, when I move an e-mail from Trash to a different folder, it even then pops up as news mail on top, be it from inside the e-mail oder from the list view.

**Comment by bansalsohil** [comment]
I am using a OS X 10.11.1 and gmail as email service provider.

**Comment by enoversum** [comment]
This is a bit frustrating. I can see those unread e-mails even in other e-mail clients, meaning N1 seems to really mark them as unread, so that they turn up in the inbox as unread mail again. I can reproduce it with each e-mail I answer, whether I archive it or not.

**Comment by mbilker** [comment]
Does this still occur?

**Comment by bansalsohil** [comment]
No, the issue is resolved thanks.

**Comment by enoversum** [comment]
Looks like it is solved for me as well :).

> On Dec 27 2015, at 7:15 am, Sohil Bansal &lt;notifications@github.com&gt;
wrote:  
No, the issue is resolved thanks.  
  
&gt; On 27-Dec-2015, at 3:21 AM, Matt Bilker &lt;notifications@github.com&gt;
wrote:  
&gt;  
&gt; Does this still occur?  
&gt;  
&gt; —  
&gt; Reply to this email directly or view it on GitHub
&lt;https://github.com/nylas/N1/issues/591#issuecomment-167364940&gt;.  
&gt;  
  

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/nylas/N
1/issues/591#issuecomment-167388481).![](https://github.com/notifications/beac
on/AHz43gDRrlyZc2fZesOCgWzHO1WJiuH6ks5pT3mGgaJpZM4GvkH4.gif)

**Comment by enoversum** [comment]
Oh no, I spoke too soon. I just archived an e-mail on my second accound, looked into the archive folder and found it marked as unread. I can only then mark it read by selecting it in the list view and mark it read there, by simply opening/reading it it still stays unread.
What also still occurs is that when I replied to a message and try to move it to a folder directly from within the message, the message sticks to the inbox. I don’t know if that is related though.

**Comment by enoversum** [comment]
Well, I have to correct myself again after some tests.

1. If I open up an unread e-mail, read it and archive it right away, it still shows as unread in the Archive folder. It can only be marked as read through picking it in the list view and using the button for 'reading' it. However, if I resend it to my inbox and read it there, it is being marked as read, and stays read when I archive it (open unread mail -> return to inbox -> open again -> archive).
2. If I open an already read e-mail and archive it, it works just fine MOST OF THE TIMES, e-mail is archived and stays read. I just had a case where I archived a conversation that then showed up as unread in the Archive folder. 
3. If I open an unread e-mail and archive it to a specific folder, it sometimes moves there, sometimes sticks to the folder it was. If it moves, it still shows as read.

To sum it up: The issue does still seem to occur, and it is hard to reveal a pattern in which cases it fails. I have a stable connection without any issues, so that might just not be the problem.

**Comment by mbilker** [comment]
@enoversum Did you archive it within 0.5s?

**Comment by enoversum** [comment]
@mbilker Unfortunately no.

**Comment by mbilker** [comment]
@enoversum No, if you did then it wouldn't trigger the read. This is as per `Preferences -> General -> Reading -> Delay for marking messages as read` which the default is 0.5 seconds.

**Comment by mbilker** [comment]
@enoversum I have mine set to `Instantly` since I sometimes use keyboard shortcuts and mark them as read in a short amount of time.

**Comment by enoversum** [comment]
'Unfortunately' was to say: No, this is not the reason for it to appear as unread. I am not that fast, getting to open and archive it in below one second I suppose. Also, since I first had my archive named 'Archived' for another e-mail client, I had to move it through the folder button on the right, so it took even more time. Furthermore, it also happens with e-mails already read.
I switched it to instantly as well, maybe it helps :smile: .

**Comment by danielbayley** [comment]
+1. Mine are being marked unread all over the place as I process mail rules.

**Comment by grinich** [comment]
This should be fixed now!

**Comment by enoversum** [comment]
@grinich Hey there, the behaviour still exists: I open an e-mail, answer it, archive it, e-mail is marked as unread in the sent/archive folder. Should I do something in order to get the new behaviour? Still the same just seconds ago :(.

**Comment by enoversum** [comment]
It is not to sending only. Archiving/moving to a different folder is enough for the e-mail to re-appear as unread again.

**Comment by danielbayley** [comment]
Same for me… This should be reopened.

**Comment by mbilker** [comment]
Reopened the issue. If someone could report their specific issue to the Nylas support staff and take a screen recording of the issue occurring.

**Comment by enoversum** [comment]
@mbilker It is actually too easy to understand to be in need of a screencast of what is happening:

1. Open e-mail, new or already read.
2. Archive e-mail or move to another folder manually, be it from the e-mail list or from inside the e-mail.
3. Find it being marked unread inside the new folder.

Sorry, have no such software on my computer atm :(.

**Comment by enoversum** [comment]
@mbilker @grinich Out of the blue, today e-mails don’t get marked as read anymore when I archive them. Realised it by accident, tested it, seems to work now.

@danielbayley can you confirm it is solved now?

**Comment by enoversum** [comment]
@mbilker @grinich Weird, now when I try again, e-mails get marked as unread again, plus appear double, i.e. two identical e-mails in one conversation, sometimes even adding up to three, adding up each time I move it again, as if it were three single e-mails. I saw that already sometimes, but it disappeared later, seemingly being a glitch in the Nylas engine?! Also, e-mails sometimes take a moment to re-appear as unread after archiving/moving, pointing to a slow engine as well.

**Comment by enoversum** [comment]
I don’t wanna flood this thread with messages, but still was able to observe that a) the behaviour only applies to one of my two accounts, unfortunately the one I use the most (work account), the other seems fine, and b) duplicated messages that appear after moving disappear after a while again. Still odd, but probably another problem?

**Comment by enoversum** [comment]
@grinich @mbilker Haha, I had to use another e-mail client to actually find the last notification mail to retrieve the link to this issue. That being said, the unread e-mails when archiving/moving to another folder definitely has bettered, even though it is not completely solved as of yet. Sometimes re-opening e-mails result in e-mails not being marked as read or even being marked as unread again, no matter how often I open them, but most of the times, opening an e-mail, answering and/or archiving them right away now works. Together with the new features and improvements, N1 finally is getting really, really useful :). Now, if just the search is working for IMAP, I would possibly not have to switch back and forth to my now secondary client anymore :).

**Comment by bengotow** [comment]
Hey! Are you still running into this issue? I believe this should be resolved now, thanks to recent changes on the sync engine side.

**Comment by bansalsohil** [comment]
Issue resolved thanks.

**Comment by danielbayley** [comment]
It seems to have improved, but it still happens.

**Comment by enoversum** [comment]
@danielbayley Yes, exactly. Improved greatly, but isn’t completely gone. It happens sometimes when you receive another mail while writing one (new mail shows below without further notice). Even re-opening it doesn’t help. And a few other occasions which I cannot really tell, but I have at least 1-2 occurrences a day.

**Comment by nienque** [comment]
I've been having this issue again today. 

I moved a few threads from one iCloud folder into another and ever since, random threads in the destination folder will be marked as unread one by one. That is to say, I select the thread, it's marked as read, but a few seconds later another thread will light up. It's like whack-a-mole. Closing and opening doesn't stop it, nor does deleting ~/Library/Application Support/Nylas N1.

**Comment by enoversum** [comment]
@nienque Yes, it occurs every now and then. I suppose they are hiccups in the sync engine. The occurrences are quite different, one is that you have an open e-mail conversation that gets a new e-mail while being open, leading to the new mail staying unread no matter how often you open the conversation again. I have to manually mark it as read. Also, still sometimes archiving after reading creates an unread mail, possibly about 2-4 in a week. Overall it really got better, though.

**Comment by nienque** [comment]
'Every now and then' and 'about 2-4 in a week' is not how I would characterise my experiences so far. More like '80% of all emails that are archived, deleted or moved between folders'. Just now I moved five threads into a different folder and over the course of four minutes they were all, one by one, marked unread.

**Comment by danielbayley** [comment]
> 'Every now and then' and 'about 2-4 in a week' is not how I would characterise my experiences so far. 

Same here. It's definitely more frequent than that.

**Comment by pszczolkaM** [comment]
Just wanted to add that this is happening to me, as well. I just recently joined Nylas N1 and when I move anything to my folders, it marks them as unread again, so far 100% of the time though I have only moved around 30 messages so far.

**Comment by pszczolkaM** [comment]
Update: this seems to have now been resolved - I didn't do anything so not sure why but everything is working well now!

**Comment by enoversum** [comment]
Lately, the unread issue has been re-surfacing for me and increased a lot. I read an e-mail and archive it, and sometimes it gets marked as unread again. Or I answer a long-read e-mail, archive it and find it unread again in my archive. Also, still some of the then 'unread' again e-mails cannot be marked read by opening it, but only by ticking the checkbox and marking it as read. Since it is so frequent again, I wouldn't consider this as solved :wink: .
